File: src/main/java/uk/gov/justice/digital/hmpps/riskprofiler/model/RiskProfileChange.java
package uk.gov.justice.digital.hmpps.riskprofiler.model;

import com.fasterxml.jackson.annotation.JsonInclude;
import lombok.*;

import java.time.LocalDateTime;

@JsonInclude(JsonInclude.Include.NON_NULL)
@Builder
@AllArgsConstructor
@NoArgsConstructor
@EqualsAndHashCode
@ToString
@Data
public class RiskProfileChange {
    private ProfileMessagePayload oldProfile;
    private ProfileMessagePayload newProfile;
    private String offenderNo;
    private LocalDateTime executeDateTime;
}

File: src/main/kotlin/uk/gov/justice/digital/hmpps/riskprofiler/model/RiskProfileChange.kt
package uk.gov.justice.digital.hmpps.riskprofiler.model

import com.fasterxml.jackson.annotation.JsonInclude
import java.time.LocalDateTime

@JsonInclude(JsonInclude.Include.NON_NULL)
data class RiskProfileChange(
  val oldProfile: ProfileMessagePayload,
  val newProfile: ProfileMessagePayload,
  val offenderNo: String,
  val executeDateTime: LocalDateTime,
)
